pkg/providers/amifamily/resolver.go

```diff
@@ -122,16 +123,24 @@ func NewResolver(amiProvider Provider) *Resolver {
 // Multiple ResolvedTemplates are returned based on the instanceTypes passed in to support special AMIs for certain instance types like GPUs.
 func (r Resolver) Resolve(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType, capacityType string, options *Options) ([]*LaunchTemplate, error) {
 	amiFamily := GetAMIFamily(nodeClass.Spec.AMIFamily, options)
-	amis, err := r.amiProvider.Get(ctx, nodeClass, options)
+	if len(nodeClass.Status.AMIs) == 0 {
+		return nil, fmt.Errorf("no amis exist given constraints")
+	}
+	amis, err := r.amiProvider.Get(ctx, nodeClass)
 	if err != nil {
 		return nil, err
 	}
-	if len(amis) == 0 {
-		return nil, fmt.Errorf("no amis exist given constraints")
-	}
-	mappedAMIs := amis.MapToInstanceTypes(instanceTypes)
+	mappedAMIs := MapToInstanceTypes(instanceTypes, amis)
 	if len(mappedAMIs) == 0 {
-		return nil, fmt.Errorf("no instance types satisfy requirements of amis %v", amis)
+		var sb strings.Builder
+		ids := lo.Map(amis, func(a v1beta1.AMI, _ int) string { return a.ID })
```

Comment:
Is this a separate change to improve the error message that we should take in a separate PR? If not, can we encapsulate this error printing behavior into a function so that it doesn't leak into the core business logic